azurerm/resource_arm_batch_pool_test.go

```diff
@@ -311,7 +311,11 @@ func TestAccAzureRMBatchPool_container(t *testing.T) {
 				Check: resource.ComposeTestCheckFunc(
 					testCheckAzureRMBatchPoolExists(resourceName),
 					resource.TestCheckResourceAttr(resourceName, "container_configuration.0.type", "DockerCompatible"),
+					resource.TestCheckResourceAttr(resourceName, "container_configuration.0.container_registries.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "container_configuration.0.container_registries.0.registry_server", "myContainerRegistry.azurecr.io"),
+					resource.TestCheckResourceAttr(resourceName, "container_configuration.0.container_registries.0.user_name", "myUserName"),
 				),
+				ExpectNonEmptyPlan: true,
```

Comment:
We shouldn't need this.. i presume it is because we are reading in the password each time.